src/library/trackset/setlogfeature.cpp

```diff
@@ -136,21 +141,36 @@ void SetlogFeature::onRightClickChild(const QPoint& globalPos, QModelIndex index
 }
 
 QList<BasePlaylistFeature::IdAndLabel> SetlogFeature::createPlaylistLabels() {
-    QList<BasePlaylistFeature::IdAndLabel> playlistLabels;
+    return QList<BasePlaylistFeature::IdAndLabel>();
+}
+
+/**
+  * Purpose: When inserting or removing playlists,
+  * we require the sidebar model not to reset.
+  * This method queries the database and does dynamic insertion
+  * Use a custom model in the histroy for grouping by year
+*/
+QModelIndex SetlogFeature::constructChildModel(int selected_id) {
+    QList<TreeItem*> data_list;
```

Comment:
Ths can be moved closer to usage and you may call 
```
data_list.reserve(playlistTableModel.rowCount())
```
To avoid memory reallocation